### PR TITLE
[CE-6163] Provide information about CasC compatibility

### PR DESCRIPTION
**Pull request by Dohbedoh**
* Controllers must be associated at controller level since version 2.277.4.x. 
* `ConnectedMasterTokenProperty` is required since version 2.289.1.2.

### CODE REVIEW

File: createTeamMasterK8s.groovy [hunk 1]
```diff
@@ -92,6 +92,11 @@ String  k8sNodeSelectors = ""
 Long    k8sTerminationGracePeriodSeconds = 1200L
 String  k8sYaml = ""
 
+/**
+ * cascBundle (optional). Configuration as Code Configuration Bundle
+ */
```
Comment (kyounger) [inline createTeamMasterK8s.groovy:97]: Probably worth noting that the following two lines of code are required to be in this order.

File: createTeamMasterK8s.groovy [hunk 2]
```diff
@@ -167,6 +172,16 @@ ManagedMaster master = teamsFolder.createProject(ManagedMaster.class, teamName)
 master.setDisplayName(teamModel.getDisplayName())
 master.setDescription(masterDescription)
 
+/************************
+ * CONFIGURATION BUNDLE *
+ ************************/
+if(cascBundle?.trim()) {
+    // Note: ConnectedMasterCascProperty supported since 2.277.4.x. For earlier version, comment out the following.
```
Comment (kyounger) [inline createTeamMasterK8s.groovy:179]: Probably also worth a comment that the following two lines are required to be in the strict order they are currently.

File: createManagedMasterK8s.groovy [hunk 3]
```diff
@@ -93,7 +93,9 @@ newInstance.setDisplayName(masterDisplayName)
  * CONFIGURATION BUNDLE *
  ************************/
 if(cascBundle?.trim()) {
+    // Note: ConnectedMasterCascProperty supported since 2.277.4.x. For earlier version, comment out the following.
```
Comment (kyounger) [inline createManagedMasterK8s.groovy:96]: Probably also worth a comment that the following two lines are required to be in the strict order they are currently.

File: createTeamMasterK8s.groovy (hunk 2 above)
Comment (Dohbedoh) [reply createTeamMasterK8s.groovy:179]: You mean that the properties must be set in that specific order ? I didn't know about that. Adding it.

File: createManagedMasterK8s.groovy (hunk 3 above)
Comment (kyounger) [reply createManagedMasterK8s.groovy:97]: I didn't dig into why, but It broke for me when using them in the opposite order. /shrug

File: createTeamMasterK8s.groovy (hunk 2 above)
Comment (Dohbedoh) [reply createTeamMasterK8s.groovy]: @kyounger according to https://github.com/cloudbees/jenkins-scripts/pull/182, this should actually be token first, then the bundle.

Comment (kyounger) [reply createTeamMasterK8s.groovy]: Yes, that matches what [works for me](https://github.com/kyounger/cbci-helmfile/blob/master/values/groovy/oc-custom-groovy/master-provisioner-shim.groovy#L58-L59) and what I was indicating earlier, but maybe wasn't fully clear on.

**Pull request merged**